src/modules/75-cd/components/PipelineSteps/DeployEnvironmentEntityStep/InfrastructureEntitiesList/InfrastructureEntitiesList.tsx

```diff
@@ -109,29 +121,61 @@ export default function InfrastructureEntitiesList({
     onInfrastructureEntityUpdate()
   }
 
+  function onDragEnd(result: DropResult): void {
+    if (!result.destination) return
+
+    const sourceIndex = result.source.index
+    const destinationIndex = result.destination.index
+    if (sourceIndex === destinationIndex) return
+
+    const infraList = cloneDeep(values.infrastructures?.[environmentIdentifier]) as SelectOption[]
+    const itemToMove = infraList.splice(sourceIndex, 1)
+    infraList.splice(destinationIndex, 0, itemToMove[0])
+
+    setFieldValue(`infrastructures.${environmentIdentifier}`, infraList)
+    produce(values, draft => {
+      set(draft, `infrastructures.${environmentIdentifier}`, infraList)
+      setSelectedInfrastructures && setSelectedInfrastructures(getAllFixedInfrastructures(draft, environmentIdentifier))
+    })
```

Comment:
Update this to something like this:
```suggestion
    const updatedValues = produce(values, draft => {
    set(draft, `infrastructures.${environmentIdentifier}`, infraList)
    })
    setSelectedInfrastructures && setSelectedInfrastructures(getAllFixedInfrastructures(updatedValues, environmentIdentifier))
```